Shared/Sources/SectionItem.swift

```diff
@@ -0,0 +1,10 @@
+// Copyright © 2020 Itty Bitty Apps. All rights reserved.
+
+import Foundation
+
+protocol SectionItem: Decodable {
```

Comment:
I think the naming here made jump some mental hoops. The name 'section item' made me believe it was an item belonging to a section. However a Section Item *has* items.

Would it be more appropriate if we were to label this as just `Section`? It would follow that `HomeSectionItem` would then become `HomeSection`

Reply:
Thanks for pointing that out! I ended up renaming `SectionItem` -> `RevertSection` just to make it clear that its a section for Revert project, not UITableView section.

Reply:
I'd go even simpler, to be honest - the fully qualified name of the class will be `Module.Type`, so it's already formally  `Revert.Section`.

Reply:
@nabby26 because we have access to module name spacing we are free to name things as succinctly as possible and fully qualify the type as necessary.

The `DataSource` definition would become:

```swift
final class CollectionDataSource<Section: Revert.Section, Cell: UICollectionViewCell>
```

Reply:
Hmm I am having a trouble here. I am trying to call Section with reference to its Module, but I can choose either `Revert_iOS` or `Revert_tvOS` module. One of the file (`NewDataSource`) is shared between both modules. When I specify `Revert_iOS` in `NewDataSource`, it will not run on tvOS :(.
Also `Revert` does not come in auto type as a module.

Reply:
I'm not sure why we did that, to be honest. I wonder if it's time to make both modules just plain old `Revert`?

Reply:
Hmm I think for the time being I will just keep the name as `RevertSection` first until we modify the Module later on.